### PR TITLE
Avoid div by zero

### PR DESCRIPTION
**Pull request by huntc**
Zero division situations are now avoided by substituting a scale of 1.0 in that instance. Additionally, time scaling is more precise being now down to milliseconds.

### CODE REVIEW

File: src/linear_axis_scale.rs [hunk 1]
```diff
@@ -31,7 +31,8 @@ impl LinearScale {
         step: f32,
         labeller: Option<Rc<Labeller>>,
     ) -> LinearScale {
-        let scale = 1.0 / (range.end - range.start);
+        let delta = range.end - range.start;
+        let scale = if delta != 0.0 { 1.0 / delta } else { 1.0 };
```
Comment (avonow) [inline src/linear_axis_scale.rs:35]: `if !range.is_empty() {...} else {...}`

File: src/time_axis_scale.rs [hunk 2]
```diff
@@ -47,10 +47,11 @@ impl TimeScale {
...
         labeller: Option<Rc<Labeller>>,
     ) -> TimeScale {
-        let time_from = range.start.timestamp();
-        let time_to = range.end.timestamp();
-        let step = step.num_seconds();
-        let scale = 1.0 / (time_to - time_from) as f32;
+        let time_from = range.start.timestamp_millis();
+        let time_to = range.end.timestamp_millis();
+        let delta = time_to - time_from;
+        let scale = if delta != 0 { 1.0 / delta as f32 } else { 1.0 };
```
Comment (avonow) [inline src/time_axis_scale.rs:53]: As above

**Review by avonow — APPROVED** [review]

File: src/linear_axis_scale.rs (hunk 1 above)
Comment (huntc) [reply src/linear_axis_scale.rs:35]: We have to calculate `delta` anyhow...

Comment (avonow) [reply src/linear_axis_scale.rs:35]: We only need to calculate delta if it's not empty, so this can save the calculation

**Pull request merged**